pkg/services/datasources/service/datasource.go

```diff
@@ -528,6 +528,15 @@ func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateD
 			}
 		}
 
+		// TODO: we will eventually remove this check for moving the resource to it's separate API
+		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI {
+			s.logger.Warn("Overriding LBAC rules with stored ones. Use updateDatasourceLBACRules API to update team HTTP headers instead.")
+			previousRules := dataSource.JsonData.Get("teamHttpHeaders")
+			if previousRules != nil {
```

Comment:
You should always override. If it's nil, make sure it's unset in the `cmd.JsonData` as well :pray:

Reply:
```go
			if previousRules == nil {
				cmd.JsonData.Del("teamHttpHeaders")
			} else {
				cmd.JsonData.Set("teamHttpHeaders", previousRules)
			}
```